src/components/UntrustedSlider.tsx

```diff
@@ -0,0 +1,156 @@
+import React from "react";
+import "./Slider.css";
+import { clamp, mapRange, stepClamp } from "./utils";
+
+export interface UntrustedSliderProps {
+  defaultValue?: string;
+  formatAriaValueText?: (value: number) => string;
+  formatValueText?: (value: number) => string;
+  inline?: boolean;
+  max?: string;
+  min?: string;
+  onChange?: React.ChangeEventHandler<HTMLInputElement>;
+  step?: string;
+  value?: string;
+}
+
+const defaultFormatValueText = (value: number) => String(value);
+
+export const UntrustedSlider: React.FC<UntrustedSliderProps> = ({
+  onChange,
+  formatValueText = defaultFormatValueText,
+  formatAriaValueText = formatValueText,
+  min: controlledMin = "0",
+  max: controlledMax = "100",
+  step: controlledStep = "1",
+  defaultValue = String(
+    Math.round((Number(controlledMax) - Number(controlledMin)) / 2)
+  ),
+  value: controlledValue,
+  inline,
+}) => {
+  const [internalValue, setInternalValue] = React.useState(
+    Number(defaultValue)
+  );
+  const value = controlledValue ? Number(controlledValue) : internalValue;
+  const [min, max, step] = [controlledMin, controlledMax, controlledStep].map(
+    Number
+  );
+  const rootClassName = inline ? "Slider Slider--inline" : "Slider";
+  const startXRef = React.useRef(0);
+  const sliderTrackRef = React.useRef<HTMLDivElement>(null);
+  const sliderRef = React.useRef<HTMLDivElement>(null);
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const handleMouseMove = React.useCallback(
+    (event: MouseEvent) => {
+      const { pageX } = event;
+      if (sliderRef.current && sliderTrackRef.current && inputRef.current) {
+        const sliderTrackRect = sliderTrackRef.current.getBoundingClientRect();
+
+        const relativeValue = clamp(
+          startXRef.current,
+          sliderTrackRect.left,
+          sliderTrackRect.right
+        );
+
+        const deltaX = pageX - startXRef.current;
+        const next = stepClamp(
+          min,
+          max,
+          step,
+          mapRange(
+            clamp(
+              sliderTrackRect.left,
+              sliderTrackRect.right,
+              relativeValue + deltaX
+            ),
+            sliderTrackRect.left,
+            sliderTrackRect.right,
+            min,
+            max
+          )
+        );
+
+        Object.getOwnPropertyDescriptor(HTMLInputElement.prototype, 'value')?.set?.call(inputRef.current, String(next));
+        inputRef.current.dispatchEvent(new Event("change", { bubbles: true }));
```

Comment:
we dispatch an event here to trigger the onChange handler injected to the `input` in the JSX portion of the code.